beamer/agent/chain.py

```diff
@@ -327,38 +327,41 @@ def fill_request(request: Request, context: Context) -> None:
             request_amount=request.amount,
         )
         return
-
-    if (
-        token.functions.allowance(context.address, context.fill_manager.address).call()
-        < request.amount
-    ):
-        func = token.functions.approve(context.fill_manager.address, allowance)
+    chain_id = ChainId(w3.eth.chain_id)
+    token_address = token.address
+    retrieved_mutex = context.fill_mutexes[(chain_id, token_address)]
+    with retrieved_mutex:
```

Comment:
Nitpick: I would just call this variable `mutex`, the `retrieved_` part is unnecessary.